tests/tensorflow_autolog/test_tensorflow2_autolog.py

```diff
@@ -182,24 +182,26 @@ def input_fn(features, labels, training=True, batch_size=256):
             # The model must choose between 3 classes.
             n_classes=3,
             model_dir=dir)
-
         classifier.train(
             input_fn=lambda: input_fn(train, train_y, training=True),
             steps=500)
-        classifier.export_saved_model(dir, receiver_fn)
-
-    shutil.rmtree(dir)
-    return client.get_run(run.info.run_id)
+        if export:
+            classifier.export_saved_model(dir, receiver_fn)
+        return client.get_run(client.list_run_infos(experiment_id='0')[0].run_id)
+    finally:
+        shutil.rmtree(dir)
 
 
 @pytest.mark.large
+@pytest.mark.parametrize('export', [True, False])
 def test_tf_estimator_autolog_logs_metrics(tf_estimator_random_data_run):
     assert 'loss' in tf_estimator_random_data_run.data.metrics
     metrics = client.get_metric_history(tf_estimator_random_data_run.info.run_id, 'loss')
     assert all((x.step-1) % 100 == 0 for x in metrics)
 
 
 @pytest.mark.large
+@pytest.mark.parametrize('export', [True])
 def test_tf_estimator_autolog_model_can_load_from_artifact(tf_estimator_random_data_run):
     artifacts = client.list_artifacts(tf_estimator_random_data_run.info.run_id)
```

Comment:
Hm how does this test work? As it stands, it only pass if we have `manual_run=True` (because otherwise the run we log metrics etc & the run we log the saved model under will be different?)

Reply:
It works because the fixture `tf_estimator_random_data_run` would return the most recent run, where in both cases of `manual_run=True` and `manual_run=False` would return the run where the model is logged.